contract/src/RadicleCloudBase.sol

```diff
@@ -0,0 +1,264 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.6;
+
+import "./ImpactPoints.sol";
+
+/// @title RadicleCloudBase does on-chain accounting and emits events for operator.
+/// @dev RadicleCloudBase is an abstract contract.
+abstract contract RadicleCloudBase {
+    /// @notice Owner is needed for changing rate and withdrawing funds.
+    address public owner;
+
+    /// @dev Keep info for tracking each deployment
+    struct Deployment {
+        uint64 start;
+        uint64 expiry;
+        uint64 rate;
+        address owner;
+    }
+    /// @notice Info regading each deployment (org)
+    mapping(address => Deployment) public dep;
+
+    /// @notice Rate (price) new buyers are charged
+    uint64 public ratePerBlock;
+    /// @notice Rate (price) currently being earned
+    int64 private runningRate = 0;
+
+    /// @notice Duration of package, roughly equivalent to one month
+    uint32 public immutable duration; // 185_142
+    /// @notice Blocks to wait before being eligible for withdraw
+    uint32 public immutable withdrawWait; // 6170
+
+    /// @notice MAXBLOCK is the maximum value of uint64
+    uint64 private constant MAXBLOCK = type(uint64).max;
```

Comment:
Should be `uint256`, since `block.number` is a `uint256`

Reply:
We're keeping block number as `uint64` since it's more than sufficient. Currently a block is generated every 14 seconds, so a `uint64` could hold 3 * 10^15 days. And on operator side, we can use built-in type `uint64` in Go, without having to resort to `math/big`.

Reply:
Well, the thing is this ends up cluttering the code with casts, making it harder to read, with no real benefit... Perhaps an improvement would be to do `uint64 block = uint64(block.number)` in the `topUp` function if you want to keep this as a u64.